mantidproject/mantid
# D4 reduction basic workflow

This PR implements the first stage of the D4 reduction, from loading nexus rawdata scan in separate files, through corrections to creating diffractograms and saving them. The corrections are: dead time, bank position offsets,  relative efficiency, and normalisation to monitor/time. The result has been verified with the responsible scientist, further tests pending during continued development along D4's reduction and analysis workflow.

**To test:**

1. Code review.
2. Get the zipped calibration and benchmark result files from below.
UPDATED:
[d4.zip](https://github.com/mantidproject/mantid/files/10114627/d4.zip)

4. The following example with a part of the scan should execute with no issues:

from mantid.simpleapi import *

```
config.appendDataSearchSubDir('ILL/D4/')

runs = '387229:387230'
output_ws = 'output'
D4ILLReduction(
    Run=runs,
    OutputWorkspace=output_ws,
    NormaliseBy='Monitor',
    BankPositionOffsetsFile='dec.dec',
    ZeroPositionAngle=0.1013,
    EfficiencyCalibrationFile='effd4c.eff',
    ClearCache=True,
    ExportAscii=True
)
```

5. Verify that the overlapping regions of the `output_diffractogram_2theta` workspace is consistent within errors with provided vanadium example.

Fixes #34091 . 

Release notes and documentation updated.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
